devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -637,130 +649,269 @@ bool YarpRobotLoggerDevice::attachAll(const yarp::dev::PolyDriverList& poly)
+    loggerOption->setParameter("remote", "/testVectorCollections");
+    if (!vectorCollectionRTDataServer.initialize(loggerOption))
+    {
+        log()->error("Failed to initalize the vectorsCollectionServer", logPrefix);
+        return false;
+    }
+    char* tmp = std::getenv("YARP_ROBOT_NAME");
+    std::string metadataName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(YARP_NAME);
+    vectorCollectionRTDataServer.populateMetadata(metadataName, {std::string(tmp)});
+
+    metadataName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(TIMESTAMPS_NAME);
+    vectorCollectionRTDataServer.populateMetadata(metadataName, {TIMESTAMPS_NAME});
+
+
     const unsigned dofs = joints.size();
     m_bufferManager.setDescriptionList(joints);
+    std::string rtMetadataName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(ROBOT_DESCRIPTON_LIST);
+    vectorCollectionRTDataServer.populateMetadata(rtMetadataName, joints);
 
-    bool ok = true;
 
     // prepare the telemetry
     if (m_streamJointStates)
     {
-        ok = ok && m_bufferManager.addChannel({"joints_state::positions", {dofs, 1}, joints});
-        ok = ok && m_bufferManager.addChannel({"joints_state::velocities", {dofs, 1}, joints});
-        ok = ok && m_bufferManager.addChannel({"joints_state::accelerations", {dofs, 1}, joints});
-        ok = ok && m_bufferManager.addChannel({"joints_state::torques", {dofs, 1}, joints});
+        ok = ok && m_bufferManager.addChannel({JOINT_STATE_POSITIONS_NAME, {dofs, 1}, joints});
+        ok = ok && m_bufferManager.addChannel({JOINT_STATE_VELOCITIES_NAME, {dofs, 1}, joints});
+        ok = ok && m_bufferManager.addChannel({JOINT_STATE_ACCLERATIONS_NAME, {dofs, 1}, joints});
+        ok = ok && m_bufferManager.addChannel({JOINT_STATE_TORQUES_NAME, {dofs, 1}, joints});
+
+        // populate metadata for joint positions
+        rtMetadataName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(JOINT_STATE_POSITIONS_NAME);
+        vectorCollectionRTDataServer.populateMetadata(rtMetadataName, joints);
+
+        // populate metadata for joint velocities
+        rtMetadataName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(JOINT_STATE_VELOCITIES_NAME);
+        vectorCollectionRTDataServer.populateMetadata(rtMetadataName, joints);
+
+        // populate metadata for joint accelerations
+        rtMetadataName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(JOINT_STATE_ACCLERATIONS_NAME);
+        vectorCollectionRTDataServer.populateMetadata(rtMetadataName, joints);
+
+        // populate metadata for joint torques
+        rtMetadataName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(JOINT_STATE_TORQUES_NAME);
+        vectorCollectionRTDataServer.populateMetadata(rtMetadataName, joints);
```

Comment:
As I mentioned F2F, maybe it is possible to define a function and reduce the number of repeated calls.

One possibility could be
```cpp
bool addJointsChannelAndMetadata(const std::string& name, const std::vector<std::string>& jointNames)
{
    bool ok = m_bufferManager.addChannel({name, {jointNames.size(), 1}, jointNames});
    rtMetadataName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + name;
    ok = ok && vectorCollectionRTDataServer.populateMetadata(rtMetadataName, jointNames);
    return ok;
}
```

This also allows to add a flag to disable the realtime logging, or the saving, if needed

Reply:
Fixed in [12f1649](https://github.com/nicktrem/bipedal-locomotion-framework/commit/12f164926d7e46f74882d66b653674543accda12)